### PR TITLE
[Ops] Add AttnRes operator

### PR DESCRIPTION
**Pull request by yzhangcs**
Summary:
- Add AttnRes naive reference and fused Triton operator under `fla.ops.attnres`.
- Reference paper: Attention Residuals (https://arxiv.org/abs/2603.15031).
- Support dense residual tensors or residual tensor lists, optional return of weights/residuals, RMSNorm epsilon, and logits scale.
- Add operator tests comparing fused forward/backward against the naive implementation.
- Add the Attention Residuals citation to README.

AttnRes formula:
For residual sources `v_l` over residual-source dimension `L`, query `q`, RMSNorm scale `w`, and hidden size `D`:

```text
k_l = RMSNorm(v_l; w, rms_eps)
p_l = softmax_l(scale * <k_l, q>)
o = sum_l p_l * v_l
```

The implementation flattens non-hidden dimensions into `N`, computes softmax over `L`, and returns `o` with the original `[..., D]` shape.

Test plan:
- `python -m ruff check fla/ops/attnres tests/ops/test_attnres.py`
- `PYTHONPYCACHEPREFIX=/tmp/fla-pycache python -m compileall -q fla/ops/attnres tests/ops/test_attnres.py`
- `git diff --check -- README.md fla/ops/attnres tests/ops/test_attnres.py`
- `pre-commit run --files README.md fla/ops/attnres/__init__.py fla/ops/attnres/fused.py fla/ops/attnres/naive.py tests/ops/test_attnres.py`
- `pytest tests/ops/test_attnres.py -q` (local environment has no Triton GPU backend, so fused cases are skipped)

Breaking changes:
- None.

### CODE REVIEW

**Comment by yzhangcs** [comment]
## Benchmark: naive vs fused AttnRes on NVIDIA GB200

`python -m benchmarks.ops.run --op fused_attnres naive_attnres`

Machine: NVIDIA GB200 | CUDA 12.9 | PyTorch 2.9.0+cu129.msh

### fwd

| L | B | T | D | fused (ms) | naive (ms) | speedup |
| -- | -- | ----- | ---- | ---------- | ---------- | ------- |
|  8 | 1 |  8192 | 2048 |      0.186 |      0.845 |  **4.5×** |
|  8 | 1 | 32768 | 2048 |      0.457 |      3.185 |  **7.0×** |
| 10 | 1 |  8192 | 4096 |      0.297 |      2.002 |  **6.7×** |
| 10 | 1 | 32768 | 4096 |      1.050 |      7.791 |  **7.4×** |
| 64 | 1 |  8192 | 8192 |      2.585 |     22.869 |  **8.8×** |

### fwdbwd

| L | B | T | D | fused (ms) | naive (ms) | speedup |
| -- | -- | ----- | ---- | ---------- | ---------- | ------- |
|  8 | 1 |  8192 | 2048 |      1.098 |      3.390 |  **3.1×** |
|  8 | 1 | 32768 | 2048 |      1.725 |     12.380 |  **7.2×** |
| 10 | 1 |  8192 | 4096 |      1.152 |      7.224 |  **6.3×** |
| 10 | 1 | 32768 | 4096 |      4.285 |     28.467 |  **6.6×** |
| 64 | 1 |  8192 | 8192 |     10.956 |     82.077 |  **7.5×** |

The fused kernel is 3-9× faster than the naive PyTorch reference across the sweep, with the largest gains at high-`L` / large-`D` (where the naive `einsum + rms_norm + softmax` chain pays multiple intermediate-tensor materializations).
